tests/unit_tests/experiment_server/test_server.py

```diff
@@ -15,19 +16,16 @@
 
 async def test_receiving_event_from_cluster(
     experiment_server_ctx: AsyncContextManager[ert.experiment_server.ExperimentServer],
-    dispatcher_factory: AsyncContextManager[
-        Callable[
-            [ert.experiment_server.ExperimentServer], Awaitable[WebSocketClientProtocol]
-        ]
-    ],
 ):
     async with experiment_server_ctx as experiment_server:
         experiment = AsyncMock(Experiment)
         experiment_server.add_experiment(experiment)
 
-        async with dispatcher_factory as make_dispatcher:
-            dispatcher = await make_dispatcher(experiment_server)
+        # we need to use awaitable here due to open_timeout is "problematic"
+        # in websockets for python3.8
+        await asyncio.sleep(0)
 
+        async with connect(experiment_server._config.dispatch_uri) as dispatcher:
```

Comment:
Using this:
```
       async for dispatcher in connect(experiment_server._config.dispatch_uri, open_timeout=None):
            try:
                event = CloudEvent(
                {
                    "type": "test.event",
                    "source": "test_receiving_event_from_cluster",
                }
            )
                await dispatcher.send(to_json(event).decode())
                break
            except websockets.ConnectionClosed:
                continue
```
will not require the use of asyncio.sleep(). I am not getting any use from open_timeout yet, but my hunch is that `async for` is the way to go for this kind of robust reconnect.